blockchain/reactor_fsm.go

```diff
@@ -0,0 +1,430 @@
+package blockchain
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/p2p"
+	"github.com/tendermint/tendermint/types"
+)
+
+// Blockchain Reactor State
+type bReactorFSMState struct {
+	name string
+
+	// called when transitioning out of current state
+	handle func(*bReactorFSM, bReactorEvent, bReactorEventData) (next *bReactorFSMState, err error)
```

Comment:
What is a reason for not using a single parameter of type`bcReactorMessage` instead of two parameters?